e2e/test/scenarios/dashboard/dashboard.cy.spec.js

```diff
@@ -71,12 +71,20 @@ describe("scenarios > dashboard", () => {
       appBar().findByText("New").click();
       popover().findByText("Dashboard").should("be.visible").click();
 
+      cy.log(
+        "pressing escape should only close the entity picker modal, not the new dashboard modal",
+      );
+      modal().findByTestId("collection-picker-button").click();
+      entityPickerModal().findByText("Select a collection");
+      cy.realPress("Escape");
+      modal().findByText("New dashboard").should("be.visible");
```

Comment:
theory: if pressing escape triggers the close of the "New dashboard" modal, that might take a split second, but line 80 might run before that happens, so the test won't catch it

Reply:
maybe `modal().findByText("New dashboard").should("be.visible").wait(500).should('be.visible')`

Reply:
not sure if there's a way to avoid the dreaded `cy.wait`...

Reply:
(although, some context, the current behavior is that the "new" modal closes before the entity picker, destroying the entity picker as a consequence, so I'm pretty confident this test will pass for the correct reason)